exercises/practice/isbn-verifier/.meta/example.jl

```diff
@@ -1,28 +1,39 @@
-macro isbn_str(s)
-    ISBN(s)
-end
-
-struct ISBN <: AbstractString
-    s::AbstractString
-
-    ISBN(s) = verify(s) ? new(replace(s, "-" => "")) : throw(DomainError(s, "invalid ISBN string"))
-end
-
-Base.string(s::ISBN) = s.s
-
-function verify(s::AbstractString)
-    s = replace(s, "-" => "")
-    chars = split(s, "")
-
-    length(chars) == 10 || return false
-
-    if last(chars) == "X"
-        chars[end] = "10"
+struct ISBN
+    s::String
```

Comment:
I'm using a `String` here rather than an `AbstractString` or a generic `T` because I want `ISBN("3-598-21508-8") == ISBN(SubString("3-598-21508-8"))`. I'm not sure if that's too subtle a point to include in the exercise, though.

Maybe a good thing for the mentoring notes.

Reply:
Might have been better to keep ISBN generic and define `==(a::ISBN, b::ISBN) = value(a) == value(b)`, tho.